### PR TITLE
FIX: TypeError bug for PBS process communication in python3

### PR DESCRIPTION
**Pull request by gjcooper**
When using python3 to call the PBS status command line script there is a TypeError crash (bytes-like object not str required). To fix this I have used the interfaces.base CommandLine class.

CommandLine already uses the default encoding to decode the
stdout/stderr of the command, so take advantage of that functionality
rather than re-implementing.

First try was to implement decoding pattern seen elsewhere - then I
changed my mind and used a CommandLine object from the interfaces.base
module. (It was already imported to the module anyway)

First Attempt
-------------

Decode the stderr using the default encoding

* Fixes TypeError bug for PBS jobs running on python3.x
* Fixes issue #1959

Second Attempt
--------------

Use the nipype CommandLine class instead of Popen

Remove unused modules

Ignore exceptions in qstat CommandLine call.

### CODE REVIEW

**Comment by gjcooper** [comment]
Additionally,

I have noticed other places in the codebase that perhaps would suffer from the same bug. I could attempt fixes for these Popen calls (via a similar method) however I wouldn't have the capability to test the other batch submission systems.

In particular I have noticed there there are possible similar problems in:
* `_is_pending` in the `oar` plugin module
* `_qacct_verified_complete` and `_run_qstat` in the `sge` plugin module

There are some other places, but I would be less confident adjusting those to use CommandLine.

File: nipype/pipeline/plugins/pbs.py [hunk 1]
```diff
@@ -46,13 +45,12 @@ def __init__(self, **kwargs):
...
 
     def _is_pending(self, taskid):
-        #  subprocess.Popen requires taskid to be a string
-        proc = subprocess.Popen(["qstat", str(taskid)],
-                                stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        _, e = proc.communicate()
+        result = CommandLine('qstat {}'.format(taskid),
+                             environ=dict(os.environ),
+                             terminal_output='allatonce',
```
Comment (oesteban) [inline nipype/pipeline/plugins/pbs.py]: I'm not sure `'allatonce'` generates both `stdout` and `stderr` in the runtime object. Have you checked?

File: nipype/pipeline/plugins/pbs.py [hunk 2]
```diff
@@ -46,13 +45,12 @@ def __init__(self, **kwargs):
         super(PBSPlugin, self).__init__(template, **kwargs)
 
     def _is_pending(self, taskid):
-        #  subprocess.Popen requires taskid to be a string
-        proc = subprocess.Popen(["qstat", str(taskid)],
-                                stdout=subprocess.PIPE,
-                                stderr=subprocess.PIPE)
-        _, e = proc.communicate()
+        result = CommandLine('qstat {}'.format(taskid),
+                             environ=dict(os.environ),
```
Comment (oesteban) [inline nipype/pipeline/plugins/pbs.py]: I think you can remove this line if you want to use the default environment

File: nipype/pipeline/plugins/pbs.py (hunk 1 above)
Comment (satra) [inline nipype/pipeline/plugins/pbs.py]: `allatonce` doesn't stream but still separates `stdout` and `stderr`

**Comment by satra** [comment]
@gjcooper - i'll merge this, but it would be good to clean up the other places where you spot this.

**Pull request merged**